core/catalog.go

```diff
@@ -290,3 +292,63 @@ func (c *Catalog) Transfers(time0, time1 string) []CatalogEntry {
 	}
 	return out
 }
+
+
+// Get specific type of transfer requests according to query
+func GetRequest(query string) ([]*Item, error) {
```

Comment:
I will make general interface after implementing pull based model. By the time I will get a clear idea about how to make it general.